### DISCUSSION TITLE
Problem with axes configuration UVWABC

### DISCUSSION THREAD

**Discussion by GrblGru** [fra589/grbl-Mega-5X#360] [Q&A] [answered]
Hi Gauthier,
May I ask for your help once again?

This time I have a problem with the axis configuration UVWABC.
At the moment I am only using the U, V and W axes.
U and V can be moved in jog mode without any problems. But as soon as I try the same with the W axis,
the system hangs up. What I'm making wrong ?

Thanks, GrblGru

![Image 1](https://github.com/fra589/grbl-Mega-5X/assets/60187049/d7f1ed81-9a61-440a-a7c9-cd598440cb48)

![Image 2](https://github.com/fra589/grbl-Mega-5X/assets/60187049/7094d84a-b601-4f0a-8f7e-59f816ef3d08)

**Comment by fra589** [comment]
Hi @GrblGru,  
  
Strange problem, the config you give work well on my test board !
But only after issuing the "magic" `$RST=*` command to reset factory default after axes definition changes.
Before the `$RST=*` command,  I had similar behavior...
 
@++;
Gauthier.

**Comment by GrblGru** [comment]
Thanks, this is really strange. I have now also executed the magic command once. Now it works. I have also tested several different $$ configurations, switched on/off and reset a few times and the error has not occurred again.

Could it be that $RST=* describes some memory address that cannot be reached by the normal $xx=nn commands?
Up to now I have assumed that when using UVW, for example, the max. Rate are adjustable under $110-$112. (Normally reserved for XYZ)
But now I'm wondering what happens when I configure X and U, for example.
Could it be that the addresses for UVW are in a range that the $$ command does not list?
So e.g. for Max. Rate $117-$119

**Reply by fra589** [reply, marked as answer]
Hi @GrblGru,  
  
Due to the axes configuration changes, the parameters can be mapped differently ans so, they have differents address in flash.
Since parameters doesn't have the same length, it may happen that a parameter was read in the flash by mixing the parts of several different parameters.
In addition, flash memory also saves internal elements of Grbl which are not accessible via the $XX=nn interface but may cause interference with other parameters. 

The `$RST=*` command clears all flash memory and rewrites all parameters to their default values (defined in the default.h file) and places them at the correct address.

Axes parameters are not managed by their name (X, Y, Z, ...) but by their internal number.
Internally, Grbl uses data tables numbered from 0 to N_AXIS - 1
For example, $110 is the max rate for the axis number 0, $111 for the axis number 1, and so on, Whatever the axis name definition.
There is no axis letter reservation.
Grbl-Mega-5X can manage up to 6 axes, therefore, each series of parameters can go from 0 to 5, $110 to $115 for example for max rate whatever the letter affected to the axis.

@++;
Gauthier.

**Reply by GrblGru** [reply]
Thank you for your detailed explanation.
I've learned something new again.  :)

Thanks again
GrblGru
